frontend/src/pages/projects/screens/detail/connections/ConnectionsTableRow.tsx

```diff
@@ -28,6 +30,8 @@ const ConnectionsTableRow: React.FC<ConnectionsTableRowProps> = ({
     );
   }, [obj, connectionTypes]);
 
+  const compatibleTypes = getCompatibleTypes(Object.keys(obj.data));
```

Comment:
On my cluster's project, I had a connection without any fields set (so an empty `data`). This caused the page to through an exception since it was undefined. Right now in the `Connection` type, it's required, but I was thinking of making it optional since for creation you use `stringData` instead.

Anyways.. should we allow creation of connections with no fields?? In my upcoming add connection modal PR, i made data an optional field anyways

Reply:
I'll add more defensive code here. Looks like types may be wrong for secrets.

Reply:
@emilys314 I changed the type for `Connection` to have optional `data` and also updated this code to check not only that the key exists but also that there is a value.